### PR TITLE
[UXIT-1664] Events Page - Fix Event Schedule Days Sorting

### PR DESCRIPTION
**Pull request by mirhamasala**
## 📝 Description

This refactor abstracts `filterAndSortScheduleDays`, a utility function to filter, sort, and organize schedule days with event data, and ensures reliable UTC handling.

- **Type:** Fix + Refactor

## 🛠️ Key Changes

- **Created `filterAndSortScheduleDays`** to handle filtering and sorting of schedule days by date and events by start time, using UTC.

## 🧪 How to Test

- **Setup:** Ensure schedule data is available and includes events with valid `date` and `start` attributes.
- **Steps to Test:**
  1. Confirm that `ScheduleTabs` correctly displays sorted days and events.
- **Expected Results:** Days should appear in ascending date order, events sorted by time.
- **Additional Notes:** Ensure the displayed times and dates reflect UTC as expected.

## 📸 Screenshots

BEFORE
CleanShot 2024-10-25 at 10 58 25@2x

AFTER
CleanShot 2024-10-25 at 10 58 03@2x

## 🔖 Resources

- [date-fns UTCDate Documentation](https://date-fns.org/)

### CODE REVIEW

File: src/app/events/[slug]/utils/filterAndSortScheduleDays.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
+      const dateA = new UTCDate(a.date).getTime()
+      const dateB = new UTCDate(b.date).getTime()
+      return dateA - dateB
+    })
+
+    return sortedDays
+  }
+}
```
Comment (CharlyMartin) [inline src/app/events/[slug]/utils/filterAndSortScheduleDays.ts:28]: Nit: I've been using `compareAsc` from `'date-fns'` to compare dates asc or desc.

```suggestion
import { UTCDate } from '@date-fns/utc'
import { compareAsc } from 'date-fns'

import type { Event } from '../../types/eventType'

export function filterAndSortScheduleDays(
  schedule: NonNullable<Event['schedule']>,
) {
  {
    const daysWithEvents = schedule.days.filter((day) => day.events.length > 0)

    const daysWithEventsSortedByTime = daysWithEvents.map((day) => ({
      ...day,
      events: day.events.sort((a, b) => {
        const startA = new UTCDate(a.start)
        const startB = new UTCDate(b.start)
        return compareAsc(startA, startB)
      }),
    }))

    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
      const dateA = new UTCDate(a.date)
      const dateB = new UTCDate(b.date)
      return compareAsc(dateA, dateB)
    })

    return sortedDays
  }
}
```

File: src/app/events/[slug]/utils/filterAndSortScheduleDays.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    const daysWithEventsSortedByTime = daysWithEvents.map((day) => ({
+      ...day,
+      events: day.events.sort((a, b) => {
+        const startA = new UTCDate(a.start).getTime()
+        const startB = new UTCDate(b.start).getTime()
+        return startA - startB
+      }),
+    }))
+
+    const sortedDays = daysWithEventsSortedByTime.sort((a, b) => {
```
Comment (CharlyMartin) [inline src/app/events/[slug]/utils/filterAndSortScheduleDays.ts:20]: Nit: Best to create a new array with `.sort` as it's doesn't do it for us, unlike most Array methods.

```suggestion
    const sortedDays = [...daysWithEventsSortedByTime].sort((a, b) => {
```

**Review by CharlyMartin — APPROVED** [review]
It works! ✨

File: src/app/events/[slug]/utils/filterAndSortScheduleDays.ts (hunk 2 above)
Comment (CharlyMartin) [reply src/app/events/[slug]/utils/filterAndSortScheduleDays.ts:20]: TIL there's a new [toSorted()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSorted) method available in JavaScript, which is the same as `sort()` but doesn't update the original array!

No need to change anything, just sharing :)

Comment (mirhamasala) [reply src/app/events/[slug]/utils/filterAndSortScheduleDays.ts:21]: Oh - Amazing! Let's use it!

Comment (mirhamasala) [reply src/app/events/[slug]/utils/filterAndSortScheduleDays.ts]: Done! Thank you, @CharlyMartin!

**Pull request merged**